File: web/src/api/secret.ts
import service from '@/utils/request';

// 获取Secret列表
export const getSecretList = (params?: any) => {
    return service({
        url: '/api/v1/secret/list',
        method: 'get',
        params
    });
};

// 创建Secret
export const createSecret = (data: any) => {
    return service({
        url: '/api/v1/secret/add',
        method: 'post',
        data
    });
};

// 删除Secret
export const deleteSecret = (data: any) => {
    return service({
        url: '/api/v1/secret/delete',
        method: 'post',
        data
    });
};

// 更新Secret
export const updateSecret = (data: any) => {
    return service({
        url: '/api/v1/secret/update',
        method: 'post',
        data
    });
};

File: server/router/secret/secret.go
package secret

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type SecretRouter struct{}

func (s *SecretRouter) InitSecretRouter(PrivateGroup *gin.RouterGroup) {
	apiRouter := PrivateGroup.Group("secret").Use(middleware.OperationRecord())
	apiWithNotRecorderRouter := PrivateGroup.Group("secret")
	{
		apiRouter.POST("add", secretApi.AddSecret)       // 创建Secret
		apiRouter.POST("delete", secretApi.DeleteSecret) // 删除Secret
		apiRouter.POST("update", secretApi.UpdateSecret) // 更新Secret
	}
	{
		apiWithNotRecorderRouter.GET("list", secretApi.GetSecretList) // 获取Secret列表
	}
}
